docs/design/2020-04-17-sql-sequence.md

```diff
@@ -0,0 +1,118 @@
+# TiDB SQL SEQUENCE
+
+- Feature Name: Sequences
+- Author(s):     [AilinKid](https://github.com/ailinkid) (Lingxiang Tai)
+- Last updated:  2020-04-17
+- Discussion at: https://tidbcommunity.slack.com/archives/CQNKBDVM0
+
+## Summary
+
+“Sequence Generator” (hereinafter referred to as “Sequence”) is a database feature introduced in SQL 2003 Standard. It is defined as “a mechanism for generating continuous numerical values, which can be either internal or external objects”.
+
+Based on the definition above, the conventional auto-increment columns of MySQL/TiDB can be considered as an internal implementation of Sequence Generator. This RFC describes the design of external Sequence Generator. Unless otherwise noted, the “Sequence” described below refers to external Sequence Generator.
+
+## Motivation
+
+- Standardization: Admittedly sequence is a well-defined syntax standard in SQL 2003. Although auto-increment can be regarded as a internal sequence implementation, it hasn't supported all the external features (such as CYCLE).
+
+- Usability: A sequence can be independent of tables, shared by multiple tables and be specified with CACHE details (either specify the number of CACHE or disable CACHE), which improves usability of the product.
+
+- Business Value: Sequence is a feature widely implemented in mainstream commercial databases. Users tend to expect TiDB can syntactically support sequence to reduce migration costs who migrate their business from other databases to TiDB. 
+
+## Background
+
+TiDB auto-increment will cache a batch values by default in each instance to ensure better performance, which means auto-increment only guarantee the uniqueness under a multi-tidb-node cluster. This kind of cache mode is uncontrollable and implicit in auto-increment.
+
+But in sequence, we take it open. Users can specify whether to use cache mode or not, what the cache size should be, what's the min and max of the range like and so on. This factors make sequence more flexible, controllable and easy to use.
+
+## Syntax and Semantics
+
+TiDB sequence is borrowed from MariaDB, which is also close to DB2 because DB2 is similar to MariaDB regarding the Sequence syntax.
```

Comment:
```suggestion
TiDB sequence is borrowed from MariaDB, which is also close to DB2 because DB2 is similar to MariaDB regarding to the Sequence syntax.
```